ray_on_golem/server/services/golem/golem.py

```diff
@@ -208,6 +214,11 @@ async def _create_stack(
 
         return stack
 
+    async def _get_proposal_expiration(self, proposal: Proposal) -> timedelta:
+        await proposal.get_data()
+
+        return proposal.get_expiration_date() - datetime.now(timezone.utc)
```

Comment:
I know it is golem-core but wouldn't it be better if `get_expiration_date` be asyncio function and handle `await proposal.get_data()`?

Reply:
It depends. This function follows a pattern that exists in golem-core - helper functions expect to have `.get_data()` called elsewhere. I decided to keep it consistent instead of introducing something different.

Reply:
We decided that usage of `async` is desired as in `Proposal` we have few methods in that way.